app_test.go

```diff
@@ -258,6 +262,75 @@ func TestNewApp(t *testing.T) {
 	})
 }
 
+type eventLogger struct {
+	l *zap.Logger
+}
+
+func (l *eventLogger) LogEvent(event fxevent.Event) {
+	l.l.Info("emitted event", zap.String("event", reflect.TypeOf(event).String()))
+}
+
+func setupFxLogger(l *zap.Logger) fxevent.Logger {
+	return &eventLogger{l: l}
+}
+
+func TestSetupLogger(t *testing.T) {
+	t.Parallel()
+	t.Run("initializing custom logger", func(t *testing.T) {
+		t.Parallel()
+		app := fxtest.New(t,
+			Provide(zap.NewExample),
+			WithLogger(setupFxLogger),
+		)
+		defer app.RequireStart().RequireStop()
+		require.NoError(t, app.Err())
+		require.NoError(t, app.Start(context.Background()))
+	})
+
+	t.Run("initializing zap production logger", func(t *testing.T) {
+		t.Parallel()
+		app := fxtest.New(t,
+			Provide(zap.NewProduction),
```

Comment:
This and the development logger are the same tests as the one above, with the difference being whether they'll write to stdout or stderr. In the interest of keeping stdout/stderr as silent as possible, I would suggest omitting these two since the one above will already cover the "instantiate a logger with no dependencies" case.